tests/backup_helper.go

```diff
@@ -2932,6 +3028,310 @@ func ValidateBackup(ctx context1.Context, backupName string, orgID string, sched
 	return err
 }
 
+// ValidateBackupWithPartialSuccess validates a backup's spec's objects (resources) and volumes. resourceTypesFilter can be used to select specific types to validate (nil means all types). This function must be called after switching to the context on which `scheduledAppContexts` exists. Cluster level resources aren't validated.
+func ValidateBackupWithPartialSuccess(ctx context1.Context, backupName string, orgID string, scheduledAppContexts []*scheduler.Context, resourceTypesFilter []string, failedVolumes []string) error {
```

Comment:
@sabrarhussaini Add a check  such that whatever volumes we are sending should have failed status, and remaining volumes should have success status.